File: src/main/java/work/lclpnet/ap2/api/base/PlayerManager.java
package work.lclpnet.ap2.api.base;

import net.minecraft.server.network.ServerPlayerEntity;

import java.util.Set;

public interface PlayerManager extends Participants {

    /**
     * Adds every non-permanent spectator to the participants list.
     */
    void reset();

    void enterFinale(Set<? extends ServerPlayerEntity> finalists);

    void addPermanentSpectator(ServerPlayerEntity player);

    void removePermanentSpectator(ServerPlayerEntity player);

    void removeParticipant(ServerPlayerEntity player);

    void bind(ParticipantListener listener);
}

File: src/lib/src/main/kotlin/work/lclpnet/ap2/game/player/PlayerManager.kt
package work.lclpnet.ap2.game.player

import net.minecraft.server.level.ServerPlayer

interface PlayerManager : Participants {

    fun offer(player: ServerPlayer): Boolean

    fun startPreparation()

    fun startMiniGame()

    fun enterFinale(finalists: Set<ServerPlayer>)

    fun addPermanentSpectator(player: ServerPlayer)

    fun removePermanentSpectator(player: ServerPlayer)

    fun isPermanentSpectator(player: ServerPlayer): Boolean

    fun bind(listener: ParticipantListener?)

    fun leaveFinale()

    val isFinale: Boolean
}
